train/train_message_decoding.py

```diff
@@ -0,0 +1,95 @@
+import trl
+from datasets import load_dataset
+from message_decoding_reward_fns import answer_reward_func, format_reward_func
+from peft import LoraConfig
+from prepare_inputs import tokenize_and_inject_images
+from transformers import AutoProcessor, Qwen2VLForConditionalGeneration
+from trl import GRPOConfig, ModelConfig
+from trl.trainer import QwenGRPOTrainer
+
+print(trl.__file__)
+
+
+dataset = load_dataset("sunildkumar/message-decoding-r1")
+train_dataset = dataset["train"]
+eval_dataset = dataset["validation"]
+
+# load the model
+model_config = ModelConfig(
+    model_name_or_path="Qwen/Qwen2-VL-2B-Instruct",
+    torch_dtype="bfloat16",
+    use_peft=True,
+)
+
+model = Qwen2VLForConditionalGeneration.from_pretrained(
+    pretrained_model_name_or_path="/millcreek/home/sunil/r1_vlm/vlm-r1-message-decoding-restart/checkpoint-60",
+    torch_dtype=model_config.torch_dtype,
+    # has to be set to false for gradient checkpointing to work
+    use_cache=False,
+    # faster generation, R1-V suggestion
+    attn_implementation="flash_attention_2",
+)
+
+peft_config = LoraConfig(
+    lora_alpha=16,
+    lora_dropout=0.05,
+    r=8,
+    bias="none",
+    target_modules=["q_proj", "v_proj"],
+    task_type="CAUSAL_LM",
+)
+
+# Up the grad norm from 2 -> 10
+# Reduce beta from 0.001 -> 0.0001
+# Increase grad accumulation from 4 to 8.
+# decrease adam_beta2 from 0.999  to 0.98
```

Comment:
I think this is a good change, I've seen beta1, beta2 = (0.9, 0.95) in practice frequently